src/routes/transactions/transactions-history.imitation-transactions.controller.spec.ts

```diff
@@ -100,1393 +105,2528 @@ describe('Transactions History Controller (Unit) - Imitation Transactions', () =
-              safeAppInfo: null,
-              timestamp: 1710927685000,
-              txInfo: {
-                direction: 'OUTGOING',
-                humanDescription: null,
-                recipient: {
-                  logoUri: null,
-                  name: null,
-                  value: multisigTransfer.to,
+            {
+              conflictType: 'None',
+              transaction: {
+                executionInfo: {
+                  confirmationsRequired: 1,
+                  confirmationsSubmitted: 1,
+                  missingSigners: null,
+                  nonce: multisigTransaction.nonce,
+                  type: 'MULTISIG',
                 },
-                richDecodedInfo: null,
-                sender: {
-                  logoUri: null,
-                  name: null,
-                  value: safe.address,
+                id: `multisig_${safe.address}_${multisigTransaction.safeTxHash}`,
+                safeAppInfo: null,
+                timestamp: 1710927685000,
+                txInfo: {
+                  direction: 'OUTGOING',
+                  humanDescription: null,
+                  recipient: {
+                    logoUri: null,
+                    name: null,
+                    value: multisigTransfer.to,
+                  },
+                  richDecodedInfo: null,
+                  sender: {
+                    logoUri: null,
+                    name: null,
+                    value: safe.address,
+                  },
+                  transferInfo: {
+                    decimals: multisigToken.decimals,
+                    imitation: false,
+                    logoUri: multisigToken.logoUri,
+                    tokenAddress: multisigToken.address,
+                    tokenName: multisigToken.name,
+                    tokenSymbol: multisigToken.symbol,
+                    trusted: null,
+                    type: 'ERC20',
+                    value: multisigTransfer.value,
+                  },
+                  type: 'Transfer',
                 },
-                transferInfo: {
-                  decimals: multisigToken.decimals,
-                  imitation: false,
-                  logoUri: multisigToken.logoUri,
-                  tokenAddress: multisigToken.address,
-                  tokenName: multisigToken.name,
-                  tokenSymbol: multisigToken.symbol,
-                  trusted: null,
-                  type: 'ERC20',
-                  value: multisigTransfer.value,
-                },
-                type: 'Transfer',
+                txStatus: 'SUCCESS',
+                txHash: multisigTransaction.transactionHash,
               },
-              txStatus: 'SUCCESS',
-              txHash: multisigTransaction.transactionHash,
+              type: 'TRANSACTION',
             },
-            type: 'TRANSACTION',
-          },
-        ]);
-      });
+          ]);
+        });
+    });
+
+    it.todo('Incoming transfer "echo"');
```

Comment:
This is in preparation for the aforementioned, forthcoming PR. Considering the length of the file, I may move it to another suite depending on the implementation.